packages/expo-dev-launcher/bundle/queries/useBranchesForApp.tsx

```diff
@@ -0,0 +1,176 @@
+import format from 'date-fns/format';
+import { gql } from 'graphql-request';
+import * as React from 'react';
+import { useInfiniteQuery } from 'react-query';
+
+import { apiClient } from '../apiClient';
+import { Toast } from '../components/Toasts';
+import { useBuildInfo } from '../providers/BuildInfoProvider';
+import { queryClient } from '../providers/QueryProvider';
+import { useToastStack } from '../providers/ToastStackProvider';
+import { primeCacheWithUpdates, Update, updatesPageSize } from './useUpdatesForBranch';
+
+const query = gql`
+  query getBranches(
+    $appId: String!
+    $offset: Int!
+    $limit: Int!
+    $updatesLimit: Int!
+    $runtimeVersion: String!
+  ) {
+    app {
+      byId(appId: $appId) {
+        updateBranches(offset: $offset, limit: $limit) {
+          id
+          name
+
+          compatibleUpdates: updates(
+            offset: 0
+            limit: 1
+            filter: { runtimeVersions: [$runtimeVersion] }
+          ) {
+            id
+          }
+
+          updates: updates(offset: 0, limit: $updatesLimit) {
+            id
+            message
+            runtimeVersion
+            createdAt
+          }
+        }
+      }
+    }
+  }
+`;
+
+export type Branch = {
+  id: string;
+  name: string;
+  updates: Update[];
+};
+
+export const branchPageSize = 10;
+
+function getBranchesAsync({
+  appId,
+  page = 1,
+  runtimeVersion,
+}: {
+  appId: string;
+  page?: number;
+  runtimeVersion: string;
+}) {
+  const offset = (page - 1) * branchPageSize;
+  const variables = {
+    appId,
+    offset,
+    limit: branchPageSize,
+    updatesLimit: updatesPageSize,
+    runtimeVersion,
+  };
+
+  const branches: Branch[] = [];
+  const incompatibleBranches: Branch[] = [];
+
+  return apiClient.request(query, variables).then((response) => {
+    const updateBranches = response.app.byId.updateBranches;
+
+    updateBranches.forEach((updateBranch) => {
+      const branch: Branch = {
+        id: updateBranch.id,
+        name: updateBranch.name,
+        updates: updateBranch.updates.map((update) => {
+          return {
+            ...update,
+            createdAt: format(new Date(update.createdAt), 'MMMM d, yyyy, h:ma'),
+          };
+        }),
+      };
+
+      const hasNoUpdates = updateBranch.updates.length === 0;
+      const isCompatible = hasNoUpdates || updateBranch.compatibleUpdates.length > 0;
+
+      if (isCompatible) {
+        branches.push(branch);
+      } else {
+        incompatibleBranches.push(branch);
+      }
+
+      // side-effect: prime the cache with branches
+      primeCacheWithBranch(appId, branch);
+
+      // side-effect: prime the cache with the first paginated updates for a branch
+      primeCacheWithUpdates(appId, branch.name, branch.updates);
```

Comment:
I really like both this side effect and how nicely documented it is here 👌  clearly and succinctly was able to answer my question about why we needed the whole first page of updates in this query